src/astro/databases/base.py

```diff
@@ -201,15 +201,44 @@ def append_table(
         self,
         source_table: Table,
         target_table: Table,
-        if_conflicts: AppendConflictStrategy = "exception",
+        source_tables_cols: Optional[List[str]] = None,
+        target_tables_cols: Optional[List[str]] = None,
     ):
         """
-        Append the source table rows into a destination table.
-        The argument `if_conflicts` allows the user to define how to handle conflicts.
+        Append the source table rows into a destination table
+        The argument `conflict_strategy` allows the user to define how to handle conflicts
 
         :param source_table: Contains the rows to be appended to the target_table
         :param target_table: Contains the destination table in which the rows will be appended
-        :param if_conflicts: The strategy to be applied if there are conflicts.
+        :param conflict_strategy: Action that needs to be taken in case there is a conflict
+        :param source_tables_cols: List of columns name in source table that will be used in appending
+        :param target_tables_cols: List of columns name in target table that will be used in appending
+        """
+        return self.merge_table(
+            source_table=source_table,
+            target_table=target_table,
+            source_tables_cols=source_tables_cols,
+            target_tables_cols=target_tables_cols,
+        )
+
+    def merge_table(
+        self,
+        source_table: Table,
+        target_table: Table,
+        conflict_strategy: DBMergeConflictStrategy = "ignore",
+        merge_cols: Optional[List[str]] = None,
+        source_tables_cols: Optional[List[str]] = None,
+        target_tables_cols: Optional[List[str]] = None,
```

Comment:
The table definition has a property `columns`:
https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/table.py#L45

Why aren't we using it in these two lines? Ideally, one of the refactoring goals was to avoid having multiple function arguments that refer to the table, file, etc. Those properties should be contained in the objects of `Table`, and `File`, where possible. The exception is arguments of the transformation itself.

Reply:
@tatiana the reason for these two lists is that the source table and the target table might have different names for these columns, so we would want to make sure that we merge the correct columns.